examples/components/TodoItem.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import TodoTextInput from './TodoTextInput';
+
+
+export default class TodoItem extends Component {
+  static propTypes = {
+    todo: PropTypes.object.isRequired,
+    editTodo: PropTypes.func.isRequired,
+    deleteTodo: PropTypes.func.isRequired,
+    markTodo: PropTypes.func.isRequired
+  };
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      edit: false
+    };
+  }
+
+  onDoubleClick() {
+    this.setState({ edit: true });
+  }
+
+  onSave(id, text) {
+    if (text.length === 0) {
+      this.props.deleteTodo(id);
+    } else {
+      this.props.editTodo(id, text);
+    }
+    this.setState({ edit: false });
+  }
+
+  render() {
+    const {todo, markTodo, deleteTodo, editTodo} = this.props;
+    return (
+      <li className={classnames({
+            completed: todo.marked,
+            editing: this.state.edit
```

Comment:
I'm not really fond of this.
I'd prefer if the view to render only what is needed.